gpytorch/kernels/grid_kron_wrapper.py

```diff
@@ -0,0 +1,150 @@
+            #     covar_columns = self.base_kernel(first_item, grid, diag=False, last_dim_is_batch=True, **params)
+            #     print("covar_columns shape = ", covar_columns.shape)
+            #     print("last_dim_is_batch", last_dim_is_batch)
+            #     covar_columns = delazify(covar_columns).squeeze(-2)
+            #
+            #     if last_dim_is_batch:
+            #         covars = [ToeplitzLazyTensor(covar_columns.squeeze(-2))]
+            #     else:
+            #         covars = [ToeplitzLazyTensor(covar_columns[i : i + 1].squeeze(-2)) for i in range(n_dim)]
+            #         print("covars = ", covars)
+            else:
+                full_covar = self.base_kernel(grid, grid, last_dim_is_batch=True, **params)
+                if last_dim_is_batch:
+                    covars = [full_covar]
+                else:
+                    covars = [full_covar[i : i + 1].squeeze(0) for i in range(n_dim)]
+            if len(covars) > 1:
+                # print("calling kronecker")
+                # print([cv.evaluate().shape for cv in covars])
+                covar = KronLazyLogDet(*covars[::-1])
+            else:
+                covar = covars[0]
+
+            if not self.training:
+                self._cached_kernel_mat = covar
+            # print("fwd returning ", covar)
+            return covar
+        else:
+            return self.base_kernel(x1, x2, diag=diag, last_dim_is_batch=last_dim_is_batch)
+
+    def __call__(self, x1, x2=None, diag=False, last_dim_is_batch=False, **params):
+        # print("custom __call__")
+        x1_, x2_ = x1, x2
+
+        # Select the active dimensions
+        if self.active_dims is not None:
+            x1_ = x1_.index_select(-1, self.active_dims)
+            if x2_ is not None:
+                x2_ = x2_.index_select(-1, self.active_dims)
+
+        # Give x1_ and x2_ a last dimension, if necessary
+        if x1_.ndimension() == 1:
+            x1_ = x1_.unsqueeze(1)
+        if x2_ is not None:
+            if x2_.ndimension() == 1:
+                x2_ = x2_.unsqueeze(1)
+            if not x1_.size(-1) == x2_.size(-1):
+                raise RuntimeError("x1_ and x2_ must have the same number of dimensions!")
+
+        if x2_ is None:
+            x2_ = x1_
+
+        return self.forward(x1_, x2_, diag=diag, last_dim_is_batch=last_dim_is_batch,
+                        **params)
+
+    def interp_predictions(self, test_x, dens_list=None):
+        ## setting of latent densities ##
+        if dens_list is not None:
+            for idx, dens in enumerate(dens_list):
+                if dens is not None:
+                    self.set_latent_params(dens, idx)
+
+        k_uu = self.base_kernel(self.grid, self.grid, last_dim_is_batch=True)
+
+        left_interp_indices, left_interp_values = self._compute_grid(test_x)
+
+        ## this might be the identity, not sure.
+        right_interp_indices, right_interp_values = self._compute_grid(self.grid)
+
```

Comment:
This many empty lines will fail the linter -- maybe run `black -l 120 <file>` on your code to help clean some of this up.